frontend/ui/elements/screensaver_menu.lua

```diff
@@ -26,114 +26,144 @@ local function genMenuItem(text, setting, value, enabled_func, separator)
+                            end,
+                            callback = function(touchmenu_instance)
+                                Screensaver:setStretchLimit(touchmenu_instance)
+                            end,
+                        },
+                },
             },
             {
-                text = _("Screensaver image"),
-                keep_menu_open = true,
-                callback = function()
-                    Screensaver:chooseFile()
-                end,
+                text = _("Unlock screen delay"),
+                sub_item_table = {
+                    genMenuItem(_("Never"), "screensaver_delay", "disable"),
+                    genMenuItem(_("1 second"), "screensaver_delay", "1"),
+                    genMenuItem(_("3 seconds"), "screensaver_delay", "3"),
+                    genMenuItem(_("5 seconds"), "screensaver_delay", "5"),
+                    genMenuItem(_("Unlock with a tap"), "screensaver_delay", "tap"),
+                    genMenuItem(_("Unlock with 'exit sleep screen' gesture"), "screensaver_delay", "gesture"),
+                },
             },
+        },
+    },
+    {
+        text = _("Sleep screen message"),
+        sub_item_table = {
             {
-                text = _("Document cover"),
-                keep_menu_open = true,
+                text = _("Add custom message to sleep screen"),
+                checked_func = function()
+                    return G_reader_settings:isTrue("screensaver_show_message")
+                end,
                 callback = function()
-                    Screensaver:chooseFile(true)
+                    G_reader_settings:toggle("screensaver_show_message")
                 end,
+                separator = true,
             },
             {
-                text = _("Screensaver message"),
-                keep_menu_open = true,
+            text = _("Edit sleep screen message"),
+            enabled_func = function() 
+                if not G_reader_settings:isTrue("screensaver_show_message") then
+                    return false    
+                end
+            end,
+            keep_menu_open = true,
                 callback = function()
                     Screensaver:setMessage()
                 end,
             },
             {
-                text = _("Covers and images settings"),
+                text = _("Background fill"),
+                help_text = _([[This option will only become available, if you have selected 'Lock the screen in current state'
+                    as screensaver and have 'Sleep screen message' on.]]),
+                enabled_func = function() 
+                    if G_reader_settings:readSetting("screensaver_type") == "disable" and G_reader_settings:isTrue("screensaver_show_message") then
+                        return true    
+                    else
+                        return false
+                    end 
+                end,
```

Comment:
Also, for all these enabled_func you added, you can make them simpler by just returning the result of the test, ie. here:
```lua
enabled_func = function()
    return G_reader_settings:readSetting("screensaver_type") == "disable" and G_reader_settings:isTrue("screensaver_show_message")
end
```
will work just the same.

Reply:
will change it then.